skiko/src/jvmMain/kotlin/org/jetbrains/skiko/redrawer/LinuxOpenGLRedrawer.kt

```diff
@@ -68,23 +83,38 @@ internal class LinuxOpenGLRedrawer(
         frameDispatcher.scheduleFrame()
     }
 
-    override fun redrawImmediately() = layer.backedLayer.lockLinuxDrawingSurface {
+    override fun redrawImmediately() {
         check(!isDisposed) { "LinuxOpenGLRedrawer is disposed" }
         update(System.nanoTime())
-        it.makeCurrent(context)
-        draw()
-        it.setSwapInterval(0)
-        it.swapBuffers()
-        OpenGLApi.instance.glFinish()
-        it.setSwapInterval(swapInterval)
+        runBlocking {
+            draw(withVsync = false)
+        }
     }
 
     private fun update(nanoTime: Long) {
         layer.update(nanoTime)
     }
 
-    private fun draw() {
-        layer.inDrawScope(layer::draw)
+    private suspend fun draw(withVsync: Boolean) {
+        layer.inDrawScope {
```

Comment:
Shalln't we swap the order and do
```
inDrawThread {
   layer.inDrawScope {
```

Reply:
No, inDrawScope should be called in Swing thread, as it contain `fallback` logic